arches/app/models/tile.py

```diff
@@ -342,18 +342,24 @@ def get_tile_data(self, user_id=None):
 
         return data
 
-    def datatype_post_save_actions(self, request=None):
-        userid = None
-        if request is not None:
-            userid = str(request.user.id)
+    def ensure_userprofile_exists(self, request):
+        try:
             if hasattr(request.user, "userprofile") is not True:
                 models.UserProfile.objects.create(user=request.user)
+        except:
+            pass
+
+    def datatype_post_save_actions(self, request=None):
+        try:
+            userid = str(request.user.id)
+        except:
+            userid = None
+
         tile_data = self.get_tile_data(userid)
-        for nodeid, value in list(tile_data.items()):
+        for nodeid in tile_data.keys():
```

Comment:
This is a nice change. There are a lot of places in Arches where we should be using .keys() and .values() instead of .items().